portfolio/src/main/webapp/discussionScript.js

```diff
@@ -13,16 +19,87 @@ function displayList() {
       content.classList.add("post-content");
       content.textContent = data[i].content;
 
+      let replies = prepareReplies(data[i].replies);
+      replies.classList.add("post-replies");
 
       let singlePost = document.createElement('div');
       singlePost.classList.add("post");
 
+      let replyButton = document.createElement('button');
+      replyButton.textContent = "Reply to this post";
+
+      singlePost.appendChild(userIdentification);
       singlePost.appendChild(title);
       singlePost.appendChild(content);
+      singlePost.appendChild(replies);
+      singlePost.appendChild(replyButton);
 
       posts.appendChild(singlePost);
+
+      replyButton.addEventListener("click", () => {
+        replyButton.classList.add("hidden");
+        createReplyForm(singlePost, data[i].id);
+      });
     }
 
     document.getElementById('discussion-container').appendChild(posts);
   })
 }
+
+/**
+ *  This function creates a form that lets user reply to the comment of their choosing
+ *  and then displays it underneath the post.
+ *  @param {Object} singlePost Post object that is being replied to
+ *  @param {number} postId Id of the post being replied to
+ */
+function createReplyForm(singlePost, postId){
+
+  let replyForm = document.createElement("form");
+  replyForm.action = `/reply?postId=${postId}`;
+  replyForm.method = "POST";
+
+  let inputBox = document.createElement("textarea");
+  inputBox.name = "reply-content"
+  inputBox.rows = "2";
+  inputBox.cols = "40";
+
+  let submitButton = document.createElement("input");
+  submitButton.type = "submit";
+  submitButton.value = "Post";
+
+  replyForm.appendChild(inputBox);
+  replyForm.appendChild(submitButton);
+
+  singlePost.appendChild(replyForm);
+}
+
+/**
+ *  This function organizes the replies of a given post
+ *  @param {Object[]} singlePostReplies Array of replies to a single post
```

Comment:
To show arrays we use {!Array<!Object>}

I also think the returns here should be of type {!Element}

Reply:
Done.